### ISSUE TITLE
Misnamings and Naming Improvements Found During Migration

### ISSUE THREAD

**Issue by NicEastvillage** [RLBot/core#52]
The following list of misnamed fields and possible naming improvements was found during the migration of Beast from the East to v5. Some of them are relevant for core others may only be relevant for the [Python interface](https://github.com/VirxEC/python-interface).

GameTickPacket -> GamePacket (as decided during a dev meeting)
GameStateType -> GameState or GameStatus (GameState clashes with state setting?)
DesiredGameInfoState -> DesiredGameState
settings.loadout_config (in `bot.toml`) -> settings.loadout_file
Bot.initialize_agent() -> Bot.initialize()
script_configurations (in MatchSettings) -> script_files (though I am not sure if that is what it is)

Side note: Is `num_cars` and `num_scripts` necessary in `rlbot.toml`?

NOTE: More may be added

**Comment by VirxEC** [comment]
0. I don't remember the meeting coming to any definite conclusion on that? But idc 
0. I like `GameStatus`, `GameState` doesn't conflict with anything but the similarity might be confusing (state setting is `DesiredGameState`)
0. `DesiredGameInfoState` becoming `DesiredGameState` conflicts with the already existing `DesiredGameState`
0. Sure, might want to see what others think about the name change tho
0. Sure
0. It's not files, `MatchSettings` contains no paths to any files. There's also `player_configurations` and I'd like the 2 to be consistent
0. Sure, I forgot that tomls actually support comments properly so might as well remove them

**Comment by NicEastvillage** [comment]
Regarding `DesiredGameInfoState` -> `DesiredGameState`: Yeah, that was a bad suggestion. But I still think the name could be improved, but the content is a bit of a mixed bag so nothing seems quite right. For context:
```
table DesiredGameInfoState {
  world_gravity_z:Float;
  game_speed:Float;
  paused:Bool;
  end_match:Bool;
}
```

Regarding `script_configurations`: Okay, that makes total sense.

**Comment by VirxEC** [comment]
I agree the name is a big long winded, but I also can't come up with anything better.
